src/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -75,7 +75,7 @@
     <Copy SourceFiles="$(NativeOutputPath)$(TargetName)$(NativeBinaryExt)" DestinationFolder="$(PublishDir)" />
   </Target>
 
-  <Target Name="CopyNativePdb" Condition="'$(DebugType)' != 'None' and '$(TargetOS)' == 'Windows_NT' and '$(NativeLib)' != 'Static'" AfterTargets="Publish">
+  <Target Name="CopyNativePdb" Condition="'$(DebugType)' != 'None' and '$(TargetOS)' == 'Windows_NT' and '$(NativeLib)' != 'Static' and $(NativeCodeGen) != 'wasm'" AfterTargets="Publish">
```

Comment:
This should not be needed. `TargetOS` condition should take care of this.

`TargetOS` should not be set to Windows when building for wasm.

Reply:
In general I agree, but my best bet without manually compilation was 
```
dotnet -r win-x64 /p:NativeCodeGen=wasm
```

And the `win-x64` was used only because other options like `linux-x64` does not works. This is only partially related to #8147 since I just want nice tooling around WASM codegen and currently want live with workarounds like that.

Also I only vaguely understand your point. Does that mean that I should override `TargetOS` based on `NativeCodeGen == wasm` condition?

Reply:
Yes. TargetOS means the OS that we are compiling for. It should be `WebAssembly` or `Browser` when we are targeting WebAssembly.

I know that the wasm support was added with number of workarounds originally. We should not be growing these workarounds. They need to be cleaned up.